core/domain/question_services.py

```diff
@@ -201,3 +227,36 @@ def update_question(committer_id, question_id, change_list, commit_message):
     """
     question = apply_change_list(question_id, change_list)
     _save_question(committer_id, question, change_list, commit_message)
+
+
+def get_questions_batch(
+        collection_id, skill_ids, user_id, batch_size):
+    """Fetches a batch of questions for a user based on the provided
+    skill_ids, matching them with user skills.
+
+    Args:
+        collection_id: str. Id of the collection.
+        skill_ids: list(str). A list of skill ids.
```

Comment:
A list of skill IDs to retrieve questions for.

Reply:
how about "A list of skill IDs for the questions to be retrieved." ?

Reply:
That's fine too -- I think they're equivalent.

Reply:
Updated.